lib/core/resolvers/resolveMultiTargeting.ts

```diff
@@ -0,0 +1,132 @@
+import type { EID, Data } from "iab-openrtb/v26";
+import type { IDMatchMethod } from "iab-adcom";
+
+type TargetingResponse = {
+  ortb2: {
+    user: { eids: EID[]; data: Data[] };
+  };
+};
+
+type MultiNodeTargetingResponse = TargetingResponse & { eidSources: Set<string> };
+type Ortb2 = TargetingResponse["ortb2"];
+
+type NodeTargetingRule = {
+  // Targeting function to resolve. e.g. window.optable.node_sdk_instance.targeting('__ip__')
+  targetingFn: () => Promise<TargetingResponse>;
+  // Matcher associated to the sdk_intance
+  matcher: string;
+  // Match method, Refer to IAB OpenRTB 2.6 EID.mm - IDMatchMethod
+  // UNKNOWN = 0; NO_MATCH = 1; COOKIE_SYNC = 2; AUTHENTICATED = 3; OBSERVED = 4; INFERENCE = 5;
+  mm: IDMatchMethod;
+  // (Optional) If provided we will only pick one resolved Ortb2Response from the most prioritize matcher.
+  // Any values below 1 will be threated as ignore
+  priority?: number;
+};
+
+// Resolve multiple targeting nodes
+// If any of the targeting nodes have the 'priority' attribute, we will resolve the most prioritize node
+// Otherwise, we will aggregate all resolved targeting data
+// If multiple nodes have the same priority, we will append the eids
+// Returns the resolved Ortb2Response and the list of eid sources that were resolved
+async function resolveMultiNodeTargeting(rules: NodeTargetingRule[]): Promise<MultiNodeTargetingResponse> {
+  const hasPriorityAttr = !!rules.filter((x) => x.priority).length;
+  return hasPriorityAttr ? resolvePriorityTargeting(rules) : resolveAggregateTargeting(rules);
+}
+
+function resolveEidsFromResponse(ortb2Eids: EID[] = [], matcher: string, mm: IDMatchMethod): EID[] {
+  return ortb2Eids
+    .map(({ uids: targetingUids, source }) => {
+      if (!targetingUids) {
+        return;
+      }
+
+      const uids = targetingUids.map(({ ext, ...rest }) => rest); // Exclude 'ext'
+      return { inserter: "optable.co", matcher, mm, source, uids };
+    })
+    .filter((x) => !!x);
+}
+
+// Aggregate all resolved targeting data
+async function resolveAggregateTargeting(rules: NodeTargetingRule[]): Promise<MultiNodeTargetingResponse> {
+  const eidSources = new Set<string>();
+  const ortb2: Ortb2 = {
+    user: {
+      data: [],
+      eids: [],
+    },
+  };
+
+  function processTokens(response: TargetingResponse, matcher: string, mm: IDMatchMethod) {
+    const { data, eids } = response.ortb2?.user;
+    const resolvedEids = resolveEidsFromResponse(eids, matcher, mm);
+    ortb2.user?.data?.push(...(data ?? []));
```

Comment:
Nitpick but the optional chaining on user.data seems unnecessary given that you always initialize L54